test/unit/utils_test.js

```diff
@@ -1424,6 +1451,31 @@ describe('Utils', function() {
         memo: null
       });
     });
+
+    it("validates a challenge containing a 'client_domain' manageData operation", () => {
+      let keypair = StellarSdk.Keypair.random();
+      let clientSigningKeypair = StellarSdk.Keypair.random();
+
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        keypair,
+        "GBDIT5GUJ7R5BXO3GJHFXJ6AZ5UQK6MNOIDMPQUSMXLIHTUNR2Q5CFNF",
+        "testanchor.stellar.org",
+        300,
+        StellarSdk.Networks.TESTNET,
+        "testanchor.stellar.org",
+        null,
+        "testdomain",
+        clientSigningKeypair.publicKey()
+      );
+
+      StellarSdk.Utils.readChallengeTx(
```

Comment:
not checking return value here? or does it throw on failure?

Reply:
Yea, `readChallengeTx()` by default expects all ManageData operations' source accounts to be the server's `SIGNING_KEY`, but the `client_domain` operation uses the client's `SIGNING_KEY`, which would normally cause the `readChallengeTx()` function to throw an error.

However, I think it is probably better not to use the `buildChallengeTx()` helper so we can explicitly build a challenge tx with the `client_domain` op.